app/src/main/java/com/example/drawingApp/CustomViews/DrawingFieldView.kt

```diff
@@ -37,10 +37,24 @@ class DrawingFieldView @JvmOverloads constructor(
     private var path = Path()
     private var drawingCanvas: Canvas? = null
 
+    /**
+     * used for setting our previous bitmap from rotation, no expectation of having drawn
+     */
     fun setBitmap(bitmap: Bitmap) {
         canvasBitmap = bitmap
     }
 
+    /**
+     * sets the bitmap from some image,currently resizes the bitmap
+     */
+    fun setBitmapFromImageBitmap(imageBitmap: Bitmap, resize: Boolean = true) {
+        if (resize) {
+            canvasBitmap = Bitmap.createScaledBitmap(imageBitmap, width, height, true)
```

Comment:
Making a resize bitmap on the main thread is a no no. An imageview has many properties on how to scale/crop on its own. Use one of those instead

Reply:
implemented a scope and the ability to not resize in the function

Reply:
this View is a customView so unless I extend it as an ImageView I'm not sure if I have the ability to use those ImageView properties